src/HttpClient.js

```diff
@@ -63,7 +63,7 @@ class HttpClient {
     allowRetry=true,
     uriIndex
   }) {
-    uriIndex = uriIndex || this.uriIndex;
+    if(uriIndex === undefined) { uriIndex = this.uriIndex; }
```

Comment:
This line is unnecessary, it's cleaner to just deal with it in `BaseURI` (like you already do).